Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,605 @@
+        if (mantissa[i] != 0) {
+            break;
+        }
+    }
+
+    return result;
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalDivide(NSDecimal* result,
+                                   const NSDecimal* leftOperand,
+                                   const NSDecimal* rightOperand,
+                                   NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+static NSCalculationError _mantissaDivision(NSDecimalInternal* result, NSDecimalInternal* dividend, unsigned short divisor) {
+    // Note: All inputs have been normalized via NSDecimalNormalize
+    if (!divisor) {
+        INVALIDATE_DECIMAL((&result->decimal));
+        return NSCalculationDivideByZero;
+    }
+
+    for (int index = dividend->decimal._length - 1; index >= 0; --index) {
+        unsigned int res = static_cast<unsigned int>((dividend->_mantissa)[index]) | (result->_remainder << 16);
+        result->_remainder = res % divisor;
+        (result->_mantissa)[index] = res / divisor;
+    }
+
+    result->decimal._length = _obtainLength(result->_mantissa, dividend->decimal._length);
+
+    return NSCalculationNoError;
+}
+
+static NSCalculationError _compactMantissa(NSDecimalInternal* iDecimal, NSRoundingMode roundingMode) {
+    // Note: All inputs have been normalized via NSDecimalNormalize
+    // divide by 10 until we can compact it.
+    int exponent = 0;
+
+    NSDecimalInternal result;
+    while (iDecimal->decimal._length > NSDecimalMaxSize) {
+        exponent++;
+        NSCalculationError error = _mantissaDivision(iDecimal, iDecimal, 10);
+        if (error != NSCalculationNoError) {
+            return error;
+        }
+    }
+
+    // update the exp
+    result.decimal._exponent += exponent;
+
+    // TODO: Add Support for rounding
+    if (result._remainder != 0) {
+    }
+
+    return NSCalculationNoError;
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSString* NSDecimalString(const NSDecimal* dcm, id locale) STUB_METHOD {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Interoperable
+*/
+NSComparisonResult NSDecimalCompare(const NSDecimal* leftOperand, const NSDecimal* rightOperand) {
+    if (leftOperand == rightOperand) {
+        return NSOrderedSame;
+    }
+
+    THROW_NS_IF_FALSE(E_INVALIDARG, (leftOperand != NULL) && (rightOperand != NULL));
+
+    NSDecimal left, right;
+    NSDecimalCopy(&left, leftOperand);
+    NSDecimalCopy(&right, rightOperand);
+    NSDecimalNormalize(&left, &right, 0);
+
+    if (!left._isNegative && right._isNegative) {
+        return NSOrderedDescending;
+    }
+
+    if (left._isNegative && !right._isNegative) {
+        return NSOrderedAscending;
+    }
+
+    // same signs
+    if (left._length < right._length) {
+        return right._isNegative ? NSOrderedDescending : NSOrderedAscending;
+    }
+
+    if (left._length > right._length) {
+        return right._isNegative ? NSOrderedAscending : NSOrderedDescending;
+    }
```

Comment:
Could that cause us to lose precision?

---
In reply to: [80589284](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80589284) [](ancestors = 80589284,80589221,80588809)